source/common/http/conn_manager_impl.cc

```diff
@@ -460,6 +481,25 @@ void ConnectionManagerImpl::doConnectionClose(
   }
 }
 
+void ConnectionManagerImpl::createStartDrainTimer(std::chrono::milliseconds drain_delay) {
+  if (!codec_) {
+    stats_.named_.downstream_cx_drain_close_.inc();
+    doConnectionClose(Network::ConnectionCloseType::FlushWrite, absl::nullopt, "");
```

Comment:
A burst of connection closes for connections with no active codec could result in a burst of work since the closes won't be spread over the close interval.

Reply:
Ah, good catch. I've updated this so that the same timer is applied to either calling `doConnectionClose()` or `startDrainSequence()`.

Reply:
Ah, this was changed and then changed back. A separate conversation in [this comment thread](https://github.com/envoyproxy/envoy/pull/16201#discussion_r672610649). I believe the two comments at two different times led us to two different conclusions. 

Happy to hear additional feedback on which might be a good approach to take here.